client/tests/wait_test.py

```diff
@@ -0,0 +1,151 @@
+            label=("test","variant"),
+            features=[("test","variant")],
+            feature_lags=[],
+            description="",
+            status="READY"
+        )
+    )
+
+    client=ServingClient(host="mock_host")
+    actual = client.training_set(TRAININGSET_NAME, TRAININGSET_VARIANT).wait()
+    assert actual != None
+
+def test_timeout_feature(mocker):
+    mocker.patch(
+        'featureform.register.ResourceClient.get_feature',
+        return_value=Feature(
+            name="",
+            source=("",""),
+            entity="",
+            value_type="",
+            owner="",
+            provider="",
+            location=ResourceColumnMapping(entity="",value="",timestamp=""),
+            description="",
+            status="PENDING"
+        )
+    )
+    client=ServingClient(host="mock_host")
+    try:
+
+        features = client.features([(FEATURE_NAME, FEATURE_VARIANT)], {FEATURE_ENTITY: FEATURE_VALUE}).wait(timeout=2)
+    except ValueError as actual_error:
+        assert actual_error.args[0] == f'Waited too long for resource {FEATURE_NAME}:{FEATURE_VARIANT} to be ready'
+
+def test_failed_feature(mocker):
+    mocker.patch(
+        'featureform.register.ResourceClient.get_feature',
+        return_value=Feature(
+            name="",
+            source=("",""),
+            entity="",
+            value_type="",
+            owner="",
+            provider="",
+            location=ResourceColumnMapping(entity="",value="",timestamp=""),
+            description="",
+            status="FAILED"
+        )
+    )
+    client=ServingClient(host="mock_host")
+    try:
+        features = client.features([(FEATURE_NAME, FEATURE_VARIANT)], {FEATURE_ENTITY: FEATURE_VALUE}).wait(timeout=2)
+    except ValueError as actual_error:
+        assert actual_error.args[0] == f'Resource {FEATURE_NAME}:{FEATURE_VARIANT} status set to failed while waiting'
+
+def test_ready_feature(mocker):
+    mocker.patch(
+        'featureform.register.ResourceClient.get_feature',
+        return_value=Feature(
+            name="",
+            source=("",""),
+            entity="",
+            owner="",
+            value_type="",
+            provider="",
+            location=ResourceColumnMapping(entity="",value="",timestamp=""),
+            description="",
+            status="READY"
+        )
+    )
+    mocker.patch(
+        'featureform.serving.FeatureServer._get_features',
+        return_value=""
+    )
+
+    expected = Feature(
+            name="",
+            source=("",""),
+            entity="",
+            value_type="",
+            owner="",
+            provider="",
+            location=ResourceColumnMapping(entity="",value="",timestamp=""),
+            description="",
+            status="READY"
+        )
+    client=ServingClient(host="mock_host")
+    actual = client.features([(FEATURE_NAME, FEATURE_VARIANT)], {FEATURE_ENTITY: FEATURE_VALUE}).wait(timeout=2)
+    assert actual != None
```

Comment:
just a reference, you can do
```
assert actual != None, "MESSAGE"
```
if you want to print a message in case it fails.

Reply:
added